woocommerce/woocommerce-ios
# Category Selector: Add search bar and clear selection button

Part of #6490 


### Description

This PR updates the category selector screen with an optional search bar and clear selection buttons. These views are hidden by default and can be enabled via config.

I'm only enabling these views for the coupon edit flow at the moment, so the category list in other flows (product form, product filter) should work as before.

The clear selection button only shows up when there's any selected item, otherwise it's hidden.

The search bar is not functioning right now, the implementation for searching categories will be handled in another PR to avoid bloating this one.

### Testing instructions

- Navigate to Menu > Coupons > select a coupon and to its edit flow (ellipsis button > Edit Coupon)
- Tap Select Product Category button. Notice that the category selector now has a search bar.
- Select any item in the category list. Notice that the Clear selection button only shows up when there's any selected item.

Please feel free to test the category list in other flows: product form or product filter to confirm that the list doesn't change.

### Screenshots


https://user-images.githubusercontent.com/5533851/167067901-701b4d1b-6293-46df-894e-87c52e588f50.mp4



---
